lib/kubernetes-deploy/pod_logs.rb

```diff
@@ -0,0 +1,95 @@
+# frozen_string_literal: true
+module KubernetesDeploy
+  class PodLogs
+    def initialize(logger:, pod_name:, container_names:)
+      @logger = logger
+      @pod_name = pod_name
+      @container_names = container_names
+      @last_timestamp_seen_for_container = {}
+      @container_logs = Hash.new { |hash, key| hash[key] = [] }
+    end
+
+    def sync(kubectl)
+      latest_logs = fetch_logs(kubectl, since: @last_timestamp_seen_for_container.values.min)
+      latest_logs.each do |container_name, log_lines|
+        next if log_lines.empty?
+
+        log_lines.each do |line|
+          timestamp, msg = split_timestamped_line(line)
+          next if likely_duplicate?(timestamp, container_name)
+          @container_logs[container_name] << msg
+          @last_timestamp_seen_for_container[container_name] = timestamp
+        end
+      end
+    end
+
+    def print_latest
+      @last_printed_indexes ||= Hash.new { |hash, key| hash[key] = -1 }
+      @container_logs.each do |container_name, logs|
+        prefix = "[#{container_name}]  " if @container_names.length > 1
+        start_at = @last_printed_indexes[container_name] + 1
+        logs[start_at..-1].each do |line|
+          @logger.info "#{prefix}#{line}"
+        end
+        @last_printed_indexes[container_name] = logs.length - 1
+      end
+    end
+
+    def print_all(prevent_duplicate: true)
```

Comment:
Plans for `prevent_duplicate` later?

Reply:
Not really. I had added this based on Flo's comment to the effect that having `print_all` no-op on the second call by default is unintuitive.